internal/configure/hosts/hc_item.go

```diff
@@ -26,84 +26,210 @@ package hosts
+		defaultValue interface{} // nil means no default value
+	}
+
+	itemSet struct {
+		items    []*item
+		key2item map[string]*item
+	}
+)
+
 var (
-	itemset = comm.NewItemSet()
+	itemset = &itemSet{
+		items:    []*item{},
+		key2item: map[string]*item{},
+	}
 
-	CONFIG_HOST = itemset.Insert(
+	CONFIG_HOST = itemset.insert(
 		"host",
-		comm.REQUIRE_STRING,
+		REQUIRE_STRING,
 		false,
 		nil,
 	)
 
-	CONFIG_NAME = itemset.Insert(
+	CONFIG_NAME = itemset.insert(
 		"name",
-		comm.REQUIRE_STRING,
+		REQUIRE_STRING,
 		false,
 		func(hc *HostConfig) interface{} {
 			return hc.getString(CONFIG_HOST)
 		},
 	)
 
-	CONFIG_HOSTNAME = itemset.Insert(
+	CONFIG_HOSTNAME = itemset.insert(
 		"hostname",
-		comm.REQUIRE_STRING,
+		REQUIRE_STRING,
 		false,
 		nil,
 	)
 
-	CONFIG_SSH_HOSTNAME = itemset.Insert(
+	CONFIG_SSH_HOSTNAME = itemset.insert(
 		"ssh_hostname",
-		comm.REQUIRE_STRING,
+		REQUIRE_STRING,
 		false,
 		nil,
 	)
 
-	CONFIG_USER = itemset.Insert(
+	CONFIG_USER = itemset.insert(
 		"user",
-		comm.REQUIRE_STRING,
+		REQUIRE_STRING,
 		false,
 		func(hc *HostConfig) interface{} {
 			return utils.GetCurrentUser()
 		},
 	)
 
-	CONFIG_SSH_PORT = itemset.Insert(
+	CONFIG_SSH_PORT = itemset.insert(
 		"ssh_port",
-		comm.REQUIRE_POSITIVE_INTEGER,
+		REQUIRE_POSITIVE_INTEGER,
 		false,
 		DEFAULT_SSH_PORT,
 	)
 
-	CONFIG_PRIVATE_CONFIG_FILE = itemset.Insert(
+	CONFIG_PRIVATE_CONFIG_FILE = itemset.insert(
 		"private_key_file",
-		comm.REQUIRE_STRING,
+		REQUIRE_STRING,
 		false,
 		func(hc *HostConfig) interface{} {
 			return fmt.Sprintf("%s/.ssh/id_rsa", utils.GetCurrentHomeDir())
 		},
 	)
 
-	CONFIG_FORWARD_AGENT = itemset.Insert(
+	CONFIG_FORWARD_AGENT = itemset.insert(
 		"forward_agent",
-		comm.REQUIRE_BOOL,
+		REQUIRE_BOOL,
 		false,
 		false,
 	)
 
-	CONFIG_BECOME_USER = itemset.Insert(
+	CONFIG_BECOME_USER = itemset.insert(
 		"become_user",
-		comm.REQUIRE_STRING,
+		REQUIRE_STRING,
 		false,
 		nil,
 	)
 )
+
+func convertSlice[T any](key, value any) ([]T, error) {
+	var slice []T
+	if !utils.IsAnySlice(value) || len(value.([]any)) == 0 {
+		return slice, errno.ERR_CONFIGURE_VALUE_REQUIRES_NONEMPTY_SLICE
+	}
+	anySlice := value.([]any)
+	switch anySlice[0].(type) {
+	case T:
+		for _, element := range anySlice {
+			slice = append(slice, element.(T))
+		}
+	default:
+		return slice, errno.ERR_UNSUPPORT_CONFIGURE_VALUE_TYPE.
+			F("%s: %v", key, value)
+	}
+
+	return slice, nil
+}
+
+func (i *item) Key() string {
+	return i.key
+}
+
+func (itemset *itemSet) insert(key string, require int, exclude bool, defaultValue interface{}) *item {
+	i := &item{key, require, exclude, defaultValue}
+	itemset.key2item[key] = i
+	itemset.items = append(itemset.items, i)
+	return i
+}
+
+func (itemset *itemSet) get(key string) *item {
+	return itemset.key2item[key]
+}
+
+func (itemset *itemSet) getAll() []*item {
+	return itemset.items
+}
+
+func (itemset *itemSet) Build(key string, value interface{}) (interface{}, error) {
```

Comment:
What difference from original Build function.

Reply:
It's my problem. The hc_item.go file doesn't need any modification